src/dataflow/src/render/mod.rs

```diff
@@ -239,13 +239,27 @@ pub fn build_compute_dataflow<A: Allocate, B: ComputeReplay>(
             // Import declared sources into the rendering context.
             for (source_id, source) in dataflow.source_imports.iter() {
                 let source_key = source.with_id(*source_id);
-                let (ok, err, token) = boundary.replay(
+                let (mut ok, mut err, token) = boundary.replay(
                     source_key,
                     region,
                     &format!("{name}-{source_id}"),
                     dataflow.id,
                 );
 
+                // We do not trust `replay` to correctly advance times.
```

Comment:
😬

Reply:
Well, it turns out it doesn't.

Reply:
In fairness, it also doesn't take `as_of` as an argument, so the only reason this ever worked in the past was a fairly careful handshake with `RenderSources`.

Reply:
This has been fixed to take `as_of` as an argument. I think we should leave in these operators, as this is a moment where invariants need to be enforced, on code that we (compute) don't directly control.